store/store.go

```diff
@@ -499,6 +506,67 @@ func (s *Store[H]) get(ctx context.Context, hash header.Hash) ([]byte, error) {
 	return data, nil
 }
 
+// notifyAndAdvance will notify waiters in heightSub and advance contiguousHead
+// based on already written headers.
+func (s *Store[H]) notifyAndAdvance(ctx context.Context, headers ...H) {
+	// always inform heightSub about new headers seen
+	for _, h := range headers {
+		s.heightSub.Notify(h.Height())
+	}
+
+	currHead := s.contiguousHead.Load()
+	if currHead != nil {
+		s.advanceContiguousHead(ctx, (*currHead).Height())
+	}
+}
```

Comment:
Propose to inline this into flushLoop like before. Flushloop similalry had announcing, appending and now advancing as well. I don't see point in extracting these operations separately. They are are equal members/operations constituting flushing as it currently standing. Alternatively, we could do `appendAndNotifyAndAdvance`, but we all agree its odd.

Reply:
Helper func `getHeights[H](headers ...H) []uint64` can be a solution for that. Done.

Reply:
Thank you!